pkg/controller/authentication/pgsql.go

```diff
@@ -0,0 +1,221 @@
+package authentication
+
+//
+// Copyright 2023 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1alpha1 "github.com/IBM/ibm-iam-operator/pkg/apis/operator/v1alpha1"
+	"github.com/IBM/ibm-iam-operator/pkg/controller/shatag"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+)
+
+const pgsqlCluster = `
+apiVersion: postgresql.k8s.enterprisedb.io/v1
+kind: Cluster
+metadata:
+  name: im-store-edb
+spec:
+  imageName: {{ .ImageName }}
+  instances: 2
+  logLevel: info
+  primaryUpdateStrategy: unsupervised
+  startDelay: 120
+  stopDelay: 90
+  storage:
+    resizeInUseVolumes: true
+    size: 20Gi
+    storageClass: {{ .StorageClass }}
+  bootstrap:
+    initdb:
+      database: iam
+      owner: iam_user
```

Comment:
can we read from the config

Reply:
I agree lets not hardcode anything

Reply:
@rwhundley Yes, We are going to use this single and constant database and owner details forever.
@rashmi43 , its not about hard-coding here, we are considering it as internal and making database and owner as constant.
But yes, as you both Rob and you said, I will expose the StorageSize and number of instances to CR so that it can be customized.